### PR TITLE
[Elixir] Add concepts.csv

### PR DESCRIPTION
**Pull request by filalex77**
Related: #839 
cc @SaschaMann 

This change is [Reviewable](https://reviewable.io/reviews/exercism/v3/862)

### CODE REVIEW

**Comment by neenjaw** [comment]
This looks like a good start from the elixir side, I am unfamiliar with the csv format so thanks for pinging @SaschaMann to this!

**Comment by SleeplessByte** [comment]
Nice. You might want to set-up the directory structure so that the categories are represented.

**Comment by nathanchere** [comment]
This already covers everything I would have added from most of the exercises I broke down except for breaking down solutions into multiple files which maybe isn't an essential concept in the same vein but I think makes some of the harder core exercises like Bowling or Forth much more manageable and readable. Not sure if I should add it to the list though?

**Comment by nathanchere** [comment]
This already covers everything I would have added from most of the exercises I broke down except for explaining sigils and breaking down solutions into multiple files which maybe isn't an essential concept in the same vein but I think makes some of the harder core exercises like Bowling or Forth much more manageable and readable. Sigils I think should go in but not sure if I should add working with multiple files to the list though?

**Comment by nathanchere** [comment]
This already covers everything I would have added from most of the exercises I broke down except for explaining sigils, ways to handle state, and breaking down solutions into multiple files which maybe isn't an essential concept in the same vein but I think makes some of the harder core exercises like Bowling or Forth much more manageable and readable. Sigils and handling state I think should go in but not sure if I should add working with multiple files to the list though?

**Comment by neenjaw** [comment]
Just because I am finding the concept exercise names to be changing as I'm implmenting them, let's restrict this to only exercises that have been merged.

File: languages/elixir/reference/concepts.csv [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+modules.require,
+modules.use,
+naming,
+naming.bang,
+naming.is-prefix,
+naming.question-mark,
+protocols.data-types,
+protocols.polymorphism,
+protocols.structs,
+type-comparison-hierarchy,
```
Comment (neenjaw) [inline languages/elixir/reference/concepts.csv:78]:
```suggestion
booleans,
```

**Pull request merged**